src/ArmTemplates/Common/API/Clients/Product/ProductsClient.cs

```diff
@@ -10,19 +10,32 @@
 using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.API.Clients.Product.Responses;
 using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Constants;
 using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Templates.ProductApis;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Templates.Products;
 using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Extractor.Models;
 
 namespace Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.API.Clients.Product
 {
     public class ProductsClient : ApiClientBase, IProductsClient
     {
-        const string GetAllApiProductsRequest = "{0}/subscriptions/{1}/resourceGroups/{2}/providers/Microsoft.ApiManagement/service/{3}/apis/{4}/products?api-version={5}";
+        const string GetAllProductsLinkedToApiRequest = "{0}/subscriptions/{1}/resourceGroups/{2}/providers/Microsoft.ApiManagement/service/{3}/apis/{4}/products?api-version={5}";
+        const string GetAllProductsRequest = "{0}/subscriptions/{1}/resourceGroups/{2}/providers/Microsoft.ApiManagement/service/{3}/products?api-version={4}";
+
+        public async Task<List<ProductsTemplateResource>> GetAllAsync(ExtractorParameters extractorParameters)
+        {
+            (string azToken, string azSubId) = await this.Auth.GetAccessToken();
```

Comment:
Could we move this to the constructor of the base class instead of calling it everywhere?

Reply:
answered [here](https://github.com/Azure/azure-api-management-devops-resource-kit/pull/639#discussion_r823687195)